app/controllers/v0/sessions_controller.rb

```diff
@@ -73,10 +78,14 @@ def parse_current_loa
       LOA::MAPPING[raw_loa]
     end
 
-    def create_new_user
-      user = User.new(user_attributes)
-      user = Decorators::MviUserDecorator.new(user).create unless user.loa1?
-      user
+    def saml_user
+      @user || create_saml_user
```

Comment:
Yes, it's for memoization.

- `@current_user` represents the user found in Redis via the session token.
- `@user` represents a new user created via SAML attributes.

I will rename `@user` to `@saml_user` to make the distinction clear